modules/3d/include/opencv2/3d/volume.hpp

```diff
@@ -0,0 +1,131 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html
+
+// This code is also subject to the license terms in the LICENSE_KinectFusion.md file found in this
+// module's directory
+
+#ifndef __OPENCV_RGBD_VOLUME_H__
+#define __OPENCV_RGBD_VOLUME_H__
+
+#include "opencv2/core/affine.hpp"
+
+namespace cv
+{
+class CV_EXPORTS_W Volume
+{
+   public:
+    Volume(float _voxelSize, Matx44f _pose, float _raycastStepFactor)
+        : voxelSize(_voxelSize),
+          voxelSizeInv(1.0f / voxelSize),
+          pose(_pose),
+          raycastStepFactor(_raycastStepFactor)
+    {
+    }
+
+    virtual ~Volume(){};
+
+    CV_WRAP
+    virtual void integrate(InputArray _depth, float depthFactor, const Matx44f& cameraPose,
+                           const Matx33f& intrinsics, const int frameId = 0)               = 0;
+    virtual void integrate(InputArray _depth, InputArray _rgb, float depthFactor,
+                           const Matx44f& cameraPose, const Matx33f& intrinsics,
+                           const Matx33f& rgb_intrinsics, const int frameId = 0)                  = 0;
+    CV_WRAP
+    virtual void raycast(const Matx44f& cameraPose, const Matx33f& intrinsics,
+                         const Size& frameSize, OutputArray points, OutputArray normals) const = 0;
+    virtual void raycast(const Matx44f& cameraPose, const Matx33f& intrinsics, const Size& frameSize,
+                         OutputArray points, OutputArray normals, OutputArray colors) const    = 0;
+    CV_WRAP
+    virtual void fetchNormals(InputArray points, OutputArray _normals) const                   = 0;
+    CV_WRAP
+    virtual void fetchPointsNormals(OutputArray points, OutputArray normals) const             = 0;
+    CV_WRAP
+    virtual void reset()                                                                       = 0;
+
+    // Works for hash-based volumes only, otherwise returns 1
+    virtual int getVisibleBlocks(int /*currFrameId*/, int /*frameThreshold*/) const { return 1; }
+    virtual size_t getTotalVolumeUnits() const { return 1; }
+
+   public:
+    const float voxelSize;
+    const float voxelSizeInv;
+    const Affine3f pose;
+    const float raycastStepFactor;
+};
+
+struct CV_EXPORTS_W VolumeParams
+{
+    enum VolumeType
```

Comment:
On the macOS build this name clashes with:
https://developer.apple.com/documentation/coreservices/volumetype?changes=l_1&language=objc

Reply:
Got it, fixed